.github/workflows/release.yml

```diff
@@ -0,0 +1,65 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+
+jobs:
+  build-and-publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          profile: minimal
+          override: true
+          components: rustfmt, clippy
+
+      - name: Build
+        run: make build
+
+      - name: Run tests
+        run: make test
+
+      - name: Run linter
+        run: make lint
+
+      - name: Get version from Cargo.toml
+        id: get_ver
+        uses: dante-signal31/rust-app-version@v1.2.0
+
+      - name: Tag
+        uses: actions/github-script@v5
+        id: create-tag
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
In the tableland org we were using `TEXTILEIO_MACHINE_ACCESS_TOKEN` here.  I'm not sure if the generic `GITHUB_TOKEN` will work, but it seems like it should.